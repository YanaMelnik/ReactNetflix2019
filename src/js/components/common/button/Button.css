.button {
    display: inline-block;
    padding: 5px 40px;
    font-family: Arial, serif;
    font-size: 12px;
    border-radius: 2px;
    color: white;
    font-weight: 600;
    margin: 10px 10px 30px;
    background-color: #f34c70;
    border: none;
    text-transform: uppercase;
}