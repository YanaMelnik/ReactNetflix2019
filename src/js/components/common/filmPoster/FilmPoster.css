.film__poster {
    max-width: 20vw;
}
