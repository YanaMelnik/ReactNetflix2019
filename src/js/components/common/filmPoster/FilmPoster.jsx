import React, { Component } from 'react';
import PropTypes from 'prop-types';

class FilmPoster extends Component {
  render() {
    return (
      <img src={this.props.imgUrl} alt={this.props.filmName}/>
    );
  }
}

FilmPoster.propTypes = {
  imgUrl: PropTypes.string.isRequired,
  filmName: PropTypes.string.isRequired,

};

export default FilmPoster;
