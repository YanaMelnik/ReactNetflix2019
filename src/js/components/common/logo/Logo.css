.logo {
  font-weight: 700;
  color: #f34c70;
  margin-left: 30px;
  line-height: 50px;
}
