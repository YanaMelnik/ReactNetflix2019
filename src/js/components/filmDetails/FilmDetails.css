.film-details {
    min-height: 170px;
    background: linear-gradient(
            rgba(0, 0, 0, 0.8),
            rgba(0, 0, 0, 0.49)
    ),
    url(https://xage.ru/media/posts/2017/5/23/netflix-viewing-pattern-comedy-at-breakfast.jpg);
    background-size: cover;
    color: white;
    font-family: sans-serif;
}

.film-details__content {
    display: flex;
    justify-content: space-between;
    align-items: start;
    padding: 30px 0 30px;
}

.film-details__content-text {
    padding: 10px 30px;
}

.film-details__header {
    display: flex;
    justify-content: space-between;
    width: 200px;
    font-size: 25px;
}

.film-details__title {
    font-weight: 500;
    color: #f34c70;
}

.film-details__rate {
    border: 1px solid white;
    border-radius: 50px;
    width: 25px;
    height: 25px;
    font-size: 15px;
    padding: 5px 0 0 5px;
}

.film-details__shortDescription {
    font-size: 14px;
    font-weight: 100;
}

.film-details__numbers-details {
    width: 120px;
    display: flex;
    justify-content: space-between;
    margin: 10px 0;
    font-weight: 700;
}

.film-details__year {

}

.film-details__duration {

}

.film-details__longDescriptions {
    width: 100%;
    text-align: right;
    font-size: 14px;
}

.header {
    display: flex;
    justify-content: space-between;
}