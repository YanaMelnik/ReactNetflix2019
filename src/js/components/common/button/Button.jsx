import React, { Component } from 'react';
import PropTypes from 'prop-types';
import 'Button.css';

class Button extends Component {
  render() {
    return (
      <button
        className='button'
        onClick={this.props.action}>
        {this.props.name}
      </button>
    );
  }
}

Button.propTypes = {
  name: PropTypes.string.isRequired,
  action: PropTypes.func.isRequired,
};

export default Button;
