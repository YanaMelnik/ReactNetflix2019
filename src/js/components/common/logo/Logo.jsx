import React from 'react';

import './Logo.css';

const Logo = () => <div
    className='logo wrapper'>
    netflixroulette
  </div>;

export default Logo;
