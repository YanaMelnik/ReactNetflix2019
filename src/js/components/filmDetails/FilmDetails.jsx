import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';
import FilmPoster from '../common/filmPoster/FilmPoster';
import Logo from '../common/logo/Logo';
import Button from '../common/button/Button';

import './FilmDetails.css';

class FilmDetails extends Component {
  goToSearch(e) {
    this.props.history.push('/');
  }

  render() {
    return (
      <div className='film-details'>
        <div className='header wrapper'>
          <Logo/>
          <Button text='Search' action={ e => this.goToSearch(e) }/>
        </div>
        <div className='wrapper film-details__content'>
          <FilmPoster filmName={this.props.filmName} imgUrl={this.props.url}/>

          <div className='film-details__content-text'>
            <div className='film-details__header'>
              <span className='film-details__title'>{this.props.filmName}</span>
              <div className='film-details__rate'>
                <span>{this.props.rate}</span>
              </div>
            </div>

            <span className='film-details__shortDescription'>{this.props.shortDescription}</span>

            <div className='film-details__numbers-details'>
              <span className='film-details__year'>{this.props.yearOfIssue}</span>
              <span className='film-details__duration'>{this.props.duration} min</span>
            </div>

            <span className='film-details__longDescriptions'>{this.props.longDescriptions}</span>
          </div>
        </div>
      </div>
    );
  }
}

FilmDetails.propTypes = {
  filmName: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  rate: PropTypes.number.isRequired,
  shortDescription: PropTypes.string.isRequired,
  yearOfIssue: PropTypes.number.isRequired,
  duration: PropTypes.number.isRequired,
  longDescriptions: PropTypes.string.isRequired,
  history: PropTypes.object,
};

export default withRouter(FilmDetails);
// export default FilmDetails;
