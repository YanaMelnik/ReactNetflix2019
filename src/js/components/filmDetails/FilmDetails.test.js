import React from 'react';
import { shallow } from 'enzyme';
import FilmDetails from './FilmDetails';
import Logo from '../common/logo/Logo';
import Button from '../common/button/Button';
import FilmPoster from '../common/filmPoster/FilmPoster';

describe('FilmDetails', () => {
    let component;
    beforeEach(() => {
        component = shallow(<FilmDetails
            duration={3}
            filmName='test'
            longDescriptions='test'
            rate={7}
            shortDescription='test'
            url='test'
            yearOfIssue={1991}/>);
    });
    it('should render component correctly in case when all props defined', () => {
        expect(component).toMatchSnapshot();
        // TODO: Is it ok to add only snapshot test for dumb components?
    });

    it('should always contain header with logo and search btn', () => {
        expect(component.find(Logo)).toHaveLength(1);
        expect(component.find(Button)).toHaveLength(1);
    });

    it('should always contain film poster', () => {
        expect(component.find(FilmPoster)).toHaveLength(1);
    });
});
