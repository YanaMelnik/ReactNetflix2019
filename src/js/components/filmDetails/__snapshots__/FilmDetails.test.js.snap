// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDetails should render component correctly in case when all props defined 1`] = `
<div
  className="film-details"
>
  <div
    className="header wrapper"
  >
    <Logo />
    <Button
      action={[Function]}
      name="Search"
    />
  </div>
  <div
    className="wrapper film-details__content"
  >
    <FilmPoster
      filmName="test"
      imgUrl="test"
    />
    <div
      className="film-details__content-text"
    >
      <div
        className="film-details__header"
      >
        <span
          className="film-details__title"
        >
          test
        </span>
        <div
          className="film-details__rate"
        >
          <span>
            7
          </span>
        </div>
      </div>
      <span
        className="film-details__shortDescription"
      >
        test
      </span>
      <div
        className="film-details__numbers-details"
      >
        <span
          className="film-details__year"
        >
          1991
        </span>
        <span
          className="film-details__duration"
        >
          3
           min
        </span>
      </div>
      <span
        className="film-details__longDescriptions"
      >
        test
      </span>
    </div>
  </div>
</div>
`;
